src/io/s3_filesys.cc

```diff
@@ -883,7 +883,7 @@ void WriteStream::Run(const std::string &method,
   std::string canonical_uri;
   std::ostringstream sauth, sdate, stoken, surl, scontent;
   std::ostringstream rheader, rdata;
-  if (path_.host.find('.', 0) == std::string::npos) {
+  if (path_.host.find('.', 0) == std::string::npos && s3_region_ == "us-east-1") {
     canonical_uri = URIEncode(path_.name, false);
     canonical_headers["host"] = path_.host + ".s3.amazonaws.com";
```

Comment:
I got your point @TccccD . But I think the solution here is to change ".s3.amazonaws.com" to ".s3."+s3_region_+".amazonaws.com" . That would work for any region instead of adding this check to make it only for us-east-1.

Reply:
Please be sure to specify s3_region as per the AWS region names like cn-north-1

Reply:
Ah okay. Missed that. 

I had already added a function in the code which determines the end point for a s3 region. Could you please use that to set the host.

`path_.host + s3::getEndpoint(s3_region_)`

Reply:
I don't understand what you mean, where can I use this function?
@rahul003

Reply:
```
  if (path_.host.find('.', 0) == std::string::npos) {
    canonical_uri = URIEncode(path_.name, false);
    canonical_headers["host"] = path_.host + "." + s3::getEndpoint(s3_region_);
    surl << "https://" << canonical_headers["host"] 
          << path_.name << "?" << GetQueryMultipart(params, false);
   } else {
     canonical_uri = URIEncode("/" + path_.host + path_.name, false);
     canonical_headers["host"] = s3_endpoint_;
     surl << "https://" << s3_endpoint_ << "/" << path_.host
          << path_.name << "?" << GetQueryMultipart(params, false);
   }
```